core/src/message.rs

```diff
@@ -575,7 +566,7 @@ pub trait MessageIdGenerator {
 #[derive(Debug, Default)]
 pub struct MessageState {
     /// Collection of outgoing messages generated.
-    pub outgoing: Vec<(OutgoingMessage, FormationStatus)>,
+    pub outgoing: Vec<(Option<Payload>, Option<OutgoingMessage>)>,
```

Comment:
I had an idea to store only `OutgoingPacket`s, but there is a problem. I find it more logical to assign `MessageId` only after confirmation of sending (using direct send via `send()` or committing via `send_commit()`) so there is need to keep it together, if we collect packets. So I decided to keep these `Options`.

We are able to keep `OutgoingMessage`s with zero `MessageId` or even `Option<MessageId> and it would be a flag of sending confirmation (so we edit `produce_outgoing()`), and we will directly edit message's payload, but would it be better?

Reply:
My idea was about something similar to the following:

```rust
// Create new message
outgoing.push(OutgoingMessage::new()); // or `default()`
// Add payload
outgoing[handle].payload.append(...);
// Commit
outgoing[handle].messageId = ...;
outgoing[handle].etc = ...;
outgoing[handle].formed = true;
```

We can use some `Option` as a flag indeed.